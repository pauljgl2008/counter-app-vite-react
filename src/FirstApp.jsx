import React from 'react'
import PropTypes from 'prop-types'

const newMessage = { message: 'hola', title: 'title' };

const getResult = (a, b) => {
    return a + b;
}

export const FirstApp = ({ name = 'Paúl', title = 'No hay título', subTitle = 'No hay subtítulo' }) => {
    return (
        <>
            <p>{name}</p>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            {/* <p>{getResult(1,4)}</p> */}
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <div data-testid="test-title"> {title} </div>
            <p>Easy React</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}