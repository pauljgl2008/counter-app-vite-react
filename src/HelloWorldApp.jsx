// function App(){
//     return (<h1>Hola Mundo</h1>);
// }

// export default App
import React from 'react'

export const HelloWorldApp = () => {
  return (
    <div>Hello World App</div>
  );
}
