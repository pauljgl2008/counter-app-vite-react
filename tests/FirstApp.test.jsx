import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => { 
    test('debe de hacer match con el snapshot', ()=>{
        const title = 'Hola, soy Pa√∫l';
        const {container} = render(<FirstApp title={title}/>)
        expect(container).toMatchSnapshot();
    })
 })