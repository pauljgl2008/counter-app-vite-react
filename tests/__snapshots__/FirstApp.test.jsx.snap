// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <p>
    Paúl
  </p>
  <p>
    Hola, soy Paúl
  </p>
  <p>
    No hay subtítulo1
  </p>
  <p>
    Easy React
  </p>
</div>
`;
